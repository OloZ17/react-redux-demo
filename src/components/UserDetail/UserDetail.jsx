import React from "react";
import { object } from "prop-types";

const UserDetail = ({ activeUser }) => {
  if (activeUser.id === 0) {
    return <div>Sélectionner un utilisateur </div>;
  }

  return (
    <div>
      <h3>Détail de {activeUser.name}</h3>
      <ul>
        <li>Id : {activeUser.id}</li>
      </ul>
    </div>
  );
};

UserDetail.propTypes = {
  activeUser: object
};

export default UserDetail;
