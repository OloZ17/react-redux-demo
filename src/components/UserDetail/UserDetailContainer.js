import React, { Component } from "react";
import { object } from "prop-types";
import { connect } from "react-redux";
import { getActiveUser } from "../../redux/user";
import UserDetail from "./UserDetail";

class UserDetailContainer extends Component {
  render() {
    return <UserDetail activeUser={this.props.myActiveUser} />;
  }
}

UserDetailContainer.propTypes = {
  // activeUser: object,
  myActiveUser: object
};

const mapStateToProps = state => ({
  // activeUser: state.activeUser,
  myActiveUser: getActiveUser(state)
});

export default connect(mapStateToProps)(UserDetailContainer);
