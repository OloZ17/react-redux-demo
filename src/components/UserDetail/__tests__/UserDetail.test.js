import React from 'react';
import { render } from '@testing-library/react';

import UserDetail from '../UserDetail';

describe('Testing UserDetail() Component', () => {
  it('should take a snapshot', () => {
    const activeUser = { id: 0, name: '' };
    const { asFragment } = render(<UserDetail activeUser={activeUser} />);
    expect(asFragment(<UserDetail />)).toMatchSnapshot();
  });
});
