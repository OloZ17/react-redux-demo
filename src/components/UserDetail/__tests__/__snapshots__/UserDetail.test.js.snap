// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UserDetail() Component should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    SÃ©lectionner un utilisateur 
  </div>
</DocumentFragment>
`;
